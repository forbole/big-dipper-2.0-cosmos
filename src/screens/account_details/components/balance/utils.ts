import { formatNumber } from '@utils/format_token';
import Big from 'big.js';

export const formatBalanceData = (data: {
  available: TokenUnit;
  delegate: TokenUnit;
  unbonding: TokenUnit;
  reward: TokenUnit;
  commission?: TokenUnit;
  total?: TokenUnit;
}) => {
  const balanceChart = [
    {
      key: 'balanceAvailable',
      display: `${formatNumber(data.available.value, data.available.exponent)} ${data.available.displayDenom.toUpperCase()}`,
      value: data.available.value,
    },
    {
      key: 'balanceDelegate',
      display: `${formatNumber(data.delegate.value, data.delegate.exponent)} ${data.delegate.displayDenom.toUpperCase()}`,
      value: data.delegate.value,
    },
    {
      key: 'balanceUnbonding',
      display: `${formatNumber(data.unbonding.value, data.unbonding.exponent)} ${data.unbonding.displayDenom.toUpperCase()}`,
      value: data.unbonding.value,
    },
    {
      key: 'balanceReward',
      display: `${formatNumber(data.reward.value, data.reward.exponent)} ${data.reward.displayDenom.toUpperCase()}`,
      value: data.reward.value,
    },
  ];

  if (data.commission && Big(data.commission.value).gt(0)) {
    balanceChart.push({
      key: 'balanceCommission',
      display: `${formatNumber(data.commission.value, data.commission.exponent)} ${data.commission.displayDenom.toUpperCase()}`,
      value: data.commission.value,
    });
  }

  return balanceChart;
};
