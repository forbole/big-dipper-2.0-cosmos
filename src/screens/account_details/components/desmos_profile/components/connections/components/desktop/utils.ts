export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'network',
    width: 25,
  },
  {
    key: 'identifier',
    width: 50,
    align: 'right',
  },
  {
    key: 'creationTime',
    width: 25,
    align: 'right',
  },
];
