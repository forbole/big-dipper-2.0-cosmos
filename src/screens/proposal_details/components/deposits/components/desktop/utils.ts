export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'depositor',
    width: 40,
  },
  {
    key: 'amount',
    width: 30,
    align: 'right',
  },
  {
    key: 'time',
    width: 30,
    align: 'right',
  },
];
