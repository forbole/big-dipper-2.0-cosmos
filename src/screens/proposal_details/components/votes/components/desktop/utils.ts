export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'voter',
    // width: 34,
    width: 50,
  },
  // {
  //   key: 'votingPower',
  //   width: 33,
  //   align: 'right',
  // },
  {
    key: 'vote',
    // width: 33,
    width: 50,
    align: 'right',
  },
];
