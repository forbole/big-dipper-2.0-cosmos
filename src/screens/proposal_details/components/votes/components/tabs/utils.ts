export const tabLabels = (data: {
  yes: number;
  no: number;
  abstain: number;
  veto: number;
  notVoted: number;
}) => {
  const {
    yes,
    no,
    abstain,
    veto,
    notVoted,
  } = data;
  const total = yes + no + abstain + veto;

  return (
    [
      {
        key: 'all',
        num: total,
      },
      {
        key: 'yes',
        num: yes,
      },
      {
        key: 'no',
        num: no,
      },
      {
        key: 'veto',
        num: veto,
      },
      {
        key: 'abstain',
        num: abstain,
      },
      {
        key: 'didNotVote',
        num: notVoted,
      },
    ]
  );
};
