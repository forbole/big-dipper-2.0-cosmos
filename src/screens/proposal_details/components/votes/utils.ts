import { VoteType } from './types';

export const getVoteKey = (vote: string) => {
  const votes = {
    VOTE_OPTION_YES: 'yes',
    VOTE_OPTION_NO: 'no',
    VOTE_OPTION_NO_WITH_VETO: 'veto',
    VOTE_OPTION_ABSTAIN: 'abstain',
    NOT_VOTED: 'notVoted',
  };

  return votes[vote] || vote;
};

export const filterDataByTab = (props: {
  data: VoteType[];
  notVoted: VoteType[];
  tab: number;
}) => {
  if (props.tab === 5) {
    return props.notVoted;
  }

  return props.data.filter((x) => {
    if (props.tab === 1) {
      return x.vote === 'VOTE_OPTION_YES';
    }

    if (props.tab === 2) {
      return x.vote === 'VOTE_OPTION_NO';
    }

    if (props.tab === 3) {
      return x.vote === 'VOTE_OPTION_NO_WITH_VETO';
    }

    if (props.tab === 4) {
      return x.vote === 'VOTE_OPTION_ABSTAIN';
    }

    return true;
  });
};
