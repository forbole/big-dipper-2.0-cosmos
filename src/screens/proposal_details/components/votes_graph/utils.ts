import * as R from 'ramda';
import Big from 'big.js';
import { formatNumber } from '@utils/format_token';
import { ThemeOptions } from '@material-ui/core/styles';
import { VotesType } from './types';

type FormatGraphType = {
  data:VotesType;
  theme:ThemeOptions;
  total: Big;
}
export const formatGraphData = ({
  data, theme, total,
}: FormatGraphType) => {
  const keys = R.keys(data);
  const color = {
    0: theme.palette.custom.charts.four,
    1: theme.palette.custom.charts.one,
    2: theme.palette.custom.charts.three,
    3: theme.palette.custom.charts.two,
  };

  const formattedData = keys.map((x, i) => {
    const selectedData = data[x] as TokenUnit;
    return ({
      name: x,
      value: Big(selectedData.value).toNumber(),
      display: formatNumber(selectedData.value, selectedData.exponent),
      percentage: total.gt(0) ? (
        `${Big(selectedData.value).div(total.toString()).times(100).toFixed(2)}%`
      ) : '0%',
      color: color[i],
    });
  });

  const notEmpty = formattedData.some((x) => x.value > 0);

  if (!notEmpty) {
    formattedData.push({
      name: 'empty',
      value: 2400,
      color: theme.palette.custom.charts.zero,
      percentage: '0%',
      display: '',
    });
  }

  return formattedData;
};
