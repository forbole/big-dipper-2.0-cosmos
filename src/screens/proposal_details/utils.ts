export const getProposalType = (proposalType: string) => {
  let type = proposalType;
  if (proposalType === '/cosmos.gov.v1beta1.TextProposal') {
    type = 'textProposal';
  }

  if (proposalType === '/cosmos.params.v1beta1.ParameterChangeProposal') {
    type = 'parameterChangeProposal';
  }

  if (proposalType === '/cosmos.upgrade.v1beta1.SoftwareUpgradeProposal') {
    type = 'softwareUpgradeProposal';
  }

  if (proposalType === '/cosmos.upgrade.v1beta1.CommunityPoolSpendProposal') {
    type = 'communityPoolSpendProposal';
  }

  return type;
};

export const shouldShowData = (status: string) => (
  [
    'PROPOSAL_STATUS_VOTING_PERIOD',
    'PROPOSAL_STATUS_PASSED',
    'PROPOSAL_STATUS_REJECTED',
    'PROPOSAL_STATUS_FAILED',
  ].includes(status)
);
