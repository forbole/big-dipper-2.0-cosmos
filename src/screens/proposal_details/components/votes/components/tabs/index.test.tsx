import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import TabsHeader from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  Search: (props) => <div id="Search" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('screen: Validators/Tabs', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <TabsHeader
          {...{
            tab: 0,
            handleTabChange: jest.fn(),
            data: {
              yes: 10,
              no: 4,
              abstain: 0,
              veto: 10,
              notVoted: 0,
            },
          }}
        />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
