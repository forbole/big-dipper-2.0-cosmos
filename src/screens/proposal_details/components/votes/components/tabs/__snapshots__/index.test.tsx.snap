// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Validators/Tabs matches snapshot 1`] = `
<div
  className="makeStyles-root"
>
  <div
    className="MuiTabs-root"
  >
    <div
      className="MuiTabs-scrollable"
      style={
        Object {
          "height": 99,
          "overflow": "scroll",
          "position": "absolute",
          "top": -9999,
          "width": 99,
        }
      }
    />
    <div
      className="MuiTabs-scroller MuiTabs-scrollable"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        className="MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-controls="simple-tabpanel-0"
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-selected"
          disabled={false}
          id="simple-tab-0"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            proposals:all (24)
          </span>
          <span
            className="PrivateTabIndicator-root PrivateTabIndicator-colorSecondary MuiTabs-indicator"
            style={Object {}}
          />
        </button>
        <button
          aria-controls="simple-tabpanel-1"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          id="simple-tab-1"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            proposals:yes (10)
          </span>
        </button>
        <button
          aria-controls="simple-tabpanel-2"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          id="simple-tab-2"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            proposals:no (4)
          </span>
        </button>
        <button
          aria-controls="simple-tabpanel-3"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          id="simple-tab-3"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            proposals:veto (10)
          </span>
        </button>
        <button
          aria-controls="simple-tabpanel-4"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          id="simple-tab-4"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            proposals:abstain (0)
          </span>
        </button>
        <button
          aria-controls="simple-tabpanel-5"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          id="simple-tab-5"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            proposals:didNotVote (0)
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
